src/Challenge.js

```diff
@@ -0,0 +1,160 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import './style/Challenge.css';
+import './style/styles.css';
+import Button from 'react-button-component'
+import Header from './components/header'
+import DifficultyTab from './DifficultyTab'
+import ShowSpinner from './components/spinner';
+
+import { 
+  fetchAllChapterNames, 
+  fetchAllPairs, 
+  fetchCompletedLessons,
+  fetchLessonById
+} from './actions/learn'
+
+import { getCurrentLessonForUser } from './actions/homepage';
+
+const CustomButton = Button.extend`
+    margin-top:4vh;
+    height: 82px;	
+    width: 270px;	
+    border: 5px solid #F5A623;	
+    border-radius: 10px;	
+    background-color: #FFFFFF;
+    font-size:30px;
+
+`
+
+
+class Challenge extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { 
+      headerLinks: ["Games", "Learn", "Home"],
+      isEzActive:false,
+      isMidActive:false,
+      isHardActive: false
+    }
+    this.BeginChallenge=this.BeginChallenge.bind(this)
```

Comment:
space between the equals ` this.BeginChallenge = this.BeginChallenge.bind(this)`